processing/src/main/java/io/druid/query/BySegmentQueryRunner.java

```diff
@@ -51,7 +51,7 @@ public BySegmentQueryRunner(
   @SuppressWarnings("unchecked")
   public Sequence<T> run(final Query<T> query, Map<String, Object> responseContext)
   {
-    if (BaseQuery.getContextBySegment(query, false)) {
+    if (QueryContexts.isBySegment(query, false)) {
```

Comment:
isBySegment is always called with false as the default argument, maybe remove the parameter